src/openforms/tests/test_session_expiry.py

```diff
@@ -0,0 +1,211 @@
+                        ),
+                        "status": 403,
+                        "detail": _(
+                            "You do not have permission to perform this action."
+                        ),
+                        "instance": "urn:uuid:95a55a81-d316-44e8-b090-0519dd21be5f",
+                    },
+                )
+
+
+@override_settings(CACHES=SESSION_CACHES, SESSION_CACHE_ALIAS="session")
+class AdminSessionExpiryTests(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        config = GlobalConfiguration.get_solo()
+        config.admin_session_timeout = 5  # minimum value
+        config.save()
+
+        # Just a random admin url to use for testing
+        cls.url = reverse("admin:submissions_submission_changelist")
+        cls.superuser = SuperUserFactory.create()
+
+    def setUp(self):
+        super().setUp()
+        self.client.force_login(self.superuser)
+
+    def test_admin_activity_session_timeout(self):
+        """
+        Assert that any activity outside of the expiry period returns a 302
+        and redirects the admin to the login page
+        """
+        # make a request to the admin to validate this modifies the session
+        with self.subTest(part="initial check"):
+            with freeze_time("2021-07-29T14:00:00Z"):
+                response = self.client.get(self.url, user=self.superuser)
+
+                self.assertEqual(response.status_code, status.HTTP_200_OK)
+                session = response.wsgi_request.session
+                self.assertEqual(session.get_expiry_age(), 300)
+                expected_expiry = datetime(2021, 7, 29, 14, 5).replace(
+                    tzinfo=timezone.utc
+                )
+                self.assertEqual(session.get_expiry_date(), expected_expiry)
+
+        # make another request to the admin to validate the session has not expired yet
+        with self.subTest(part="check not expired"):
+            with freeze_time("2021-07-29T14:04:00Z"):
+                response = self.client.get(self.url, user=self.superuser)
+
+                self.assertEqual(response.status_code, status.HTTP_200_OK)
+                session = response.wsgi_request.session
+                self.assertEqual(session.get_expiry_age(), 300)
+                expected_expiry = datetime(2021, 7, 29, 14, 9).replace(
+                    tzinfo=timezone.utc
+                )
+                self.assertEqual(session.get_expiry_date(), expected_expiry)
+
+        # make another request to the admin to validate the session has expired
+        #  and the user will be redirected to the admin login page
+        with self.subTest(part="check expired and redirection to login page"):
+            with freeze_time("2021-07-29T14:10:00Z"):
+                response = self.client.get(self.url, user=self.superuser)
+
+                self.assertEqual(response.status_code, status.HTTP_302_FOUND)
```

Comment:
This should rather use `self.assertRedirects`

Reply:
Okay, good to know to next time :)